src/confluent_kafka/admin/__init__.py

```diff
@@ -1010,5 +1056,48 @@ def alter_user_scram_credentials(self, alterations, **kwargs):
                                                  AdminClient._make_user_scram_credentials_result)
 
         super(AdminClient, self).alter_user_scram_credentials(alterations, f, **kwargs)
+        return futmap
+
+    def list_offsets(self, topic_partition_offsets, **kwargs):
+        """
+        Enables to find the beginning offset,
+        end offset as well as the offset matching a timestamp
+        or the offset with max timestamp in partitions.
+
+        :param dict([TopicPartition, OffsetSpec]) topic_partition_offsets: Dictionary of
+               TopicPartition objects associated with the corresponding OffsetSpec to query for.
+        :param IsolationLevel isolation_level: The isolation level to use when
+               querying.
+        :param float request_timeout: The overall request timeout in seconds,
+               including broker lookup, request transmission, operation time
+               on broker, and response. Default: `socket.timeout.ms*1000.0`
+
+        :returns: A dict of futures keyed by TopicPartition.
+                  The future result() method returns ListOffsetsResultInfo
+                  raises KafkaException
+
+        :rtype: dict[TopicPartition, future]
+
+        :raises TypeError: Invalid input type.
+        :raises ValueError: Invalid input value.
+        """
+        AdminClient._check_list_offsets_request(topic_partition_offsets)
+
+        if 'isolation_level' in kwargs:
+            if not isinstance(kwargs['isolation_level'], _IsolationLevel):
+                raise TypeError("isolation_level argument should be an IsolationLevel")
+            kwargs['isolation_level'] = kwargs['isolation_level'].value
+
+        topic_partition_offsets_copy = []
+        for topic_partition, offset_spec in topic_partition_offsets.items():
+            topic_partition_offsets_copy.append(_TopicPartition(topic_partition.topic, int(topic_partition.partition),
+                                                int(offset_spec._value)))
+
+        topic_partition_keys = [_TopicPartition(topic_partition.topic, topic_partition.partition)
```

Comment:
Why are we not using `TopicPartition` from the input given by the user?

Reply:
Because it might have offset or other value set and we do not want that in our key so only topic and partition should be set and for the internal usecase the offset_spec is stored in the offset.

Reply:
This has been changed because the rich compare method only checks the topic and partition